File: JS/ArticleHistory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
// Caretaker
class ArticleHistory {
    constructor() {
        this.states = [];
    }
    save(state) {
        this.states.push(state);
    }
    undo() {
        if (this.states.length > 1) {
            this.states.pop();
            return this.states[this.states.length - 1];
        }
        else if (this.states.length === 1) {
            return this.states[0];
        }
        return null;
    }
    getHistory() {
        return this.states;
    }
}
exports.default = ArticleHistory;

File: Python/ArticleHistory.py
# Caretaker
class ArticleHistory:
    def __init__(self):
        self.__states = []

    def save(self, state):
        self.__states.append(state)

    def undo(self):
        if len(self.__states) > 1:
            self.__states.pop()
            return self.__states[-1]
        elif len(self.__states) == 1:
            return self.__states[0]
        return None

    def get_history(self):
        return self.__states
